doc/ref_arch/openstack/chapters/chapter04.md

```diff
@@ -139,8 +139,29 @@ Caveats:
--	Single pool of hardware resources including for controllers have the same CPU type. This is operationally efficient as any server can be utilized to support a flavour or controller. The single pool is valuable with unpredictable workloads or when the demand of certain flavours is insufficient to justify individual hardware selection. The specifications for this type of resource pooling is specified in 4.5.3.
+-	Multiple pools of hardware resources where each resource pool caters for workloads of a specific profile (for example, network intensive). Leads to efficient use of the hardware as the server resources are specific to the flavour. If not properly sized or when demand changes can lead to oversupply/starvation scenarios; reconfiguration may not be possible because of the underlying hardware or inability to vacate servers for reconfiguration to support another flavour type. 
+-	Single pool of hardware resources including for controllers have the same CPU type. This is operationally efficient as any server can be utilized to support a flavour or controller. The single pool is valuable with unpredictable workloads or when the demand of certain flavours is insufficient to justify individual hardware selection. 
+
+#### 4.2.2.7. Reservation of Compute Node Cores
+The [RA-1 2.3.2 Infrastructure Requirements](https://github.com/cntt-n/CNTT/blob/pgoyal01-patch-2/doc/ref_arch/openstack/chapters/chapter02.md#232-infrastructure-requirements) req.inf.com.08 requires the allocation of "certain number of host cores/threads to non-tenant workloads such as for OpenStack services." A number ("n") of random cores can be reserved for host services (including OpenStack services) by specifying the following in nova.conf:
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; reserved_host_cpus = n
+
+where n is any positive integer.
+
+If we wish to dedicate specific cores for host processing we need to consider two different use cases:
+
+    1. Require dedicated cores for Guest resources
+    2. No dedicated cores are required for Guest resources
+
+Scenario #1, results in compute nodes that host both pinned and unpinned workloads. In the OpenStack Pike release, scenario #1 is not supported; it may also be something that operators may not allow. Scenario #2 is supported through the specication of the cpu_shared_set configuration. The cores and their sibling threads dedicated to the host services are those that do no exist in the cpu_shared_set configuration.
```

Comment:
I suggest to change it to this:
>If we wish to dedicate specific cores for host processing we need to consider two different use cases:
>
>    1. No dedicated cores are required for Guest resources
>    2. Require dedicated cores for Guest resources
>
> Scenario #1 is supported through the specification of the cpu_shared_set configuration. The cores and their sibling threads dedicated to the host services are those that do no exist in the cpu_shared_set configuration. Scenario #2, results in compute nodes that host both pinned and unpinned workloads. In the OpenStack Pike release, Scenario #2 is not supported, however can be partially achieved by managing isolating at compute nodes level. A host aggregates can have nodes all having CPU allocation ratio as 1:1 and a pinned workload can request the placement in that host aggregate with flavor metadata. 

I swapped the order as 2 is described by 1.

Reply:
@deebhatia actually that is not the case. If we support dedicated cores for Guest resources then that also supports the case of no dedicated cores.

Reply:
Yes, I agree and for that I included **partially achieved**. We can add a note to be more explicit. 
> The host being used for pinned workload doesn't forbid the creation of unpinned workloads. It has to be controlled by the placement specs of host aggregates.

Reply:
Why does scenario 2 result in compute nodes that host both pinned and unpinned workloads?  
Should we not be explicitly preventing this situation, and mandating the use of extra_specs that ensure that when VNFs request pinned CPUs, that nothing (not unpinned workloads, not host workloads) wind up scheduled on those pinned cores?

Reply:
> Should we not be explicitly preventing this situation?

Yes.
In latest OpenStack release train, it is done by having two mutually exclusive sets for pinned and unpinned CPUs in a single compute node. This removes the need of extra_specs. In Pike, we still have to manage this out as you said.

Reply:
I agree with Pankaj modification:
reserved_host_cpu to reserve a number of CPUs for hosts services and if we want to ensure specific CPUs for host services then use vcpu_pin_set to isolate host services from tenant workloads